zips/draft-ecc-explicit-fees.rst

```diff
@@ -0,0 +1,130 @@
+::
+
+  ZIP: Unassigned
+  Title: Explicit Fees
+  Owners: Daira-Emma Hopwood <daira-emma@electriccoin.co>
+          Kris Nuttycombe <kris@electriccoin.co>
+          Jack Grigg <jack@electriccoin.co>
+  Credits: Simon Liu
+  Status: Draft
+  Category: Consensus
+  License: MIT
+  Discussions-To: <https://github.com/zcash/zips/issues/803>
+  Pull-Request: <https://github.com/zcash/zips/pull/919>
+
+
+Terminology
+===========
+
+The key word "MUST" in this document is to be interpreted as described in BCP 14 [#BCP14]_
+when, and only when, it appears in all capitals.
+
+The character § is used when referring to sections of the Zcash Protocol Specification
+[#protocol]_.
+
+
+Abstract
+========
+
+This proposal adds an explicit ``fee`` field to the v6 transaction format.
+Instead of fees being implicit in the difference between the input value and
+output value of the transaction, all value transfers, including fee transfers to
+miners, will be explicit and committed to via the txid.
+
+
+Motivation
+==========
+
+When it comes to fee selection, it should be very hard to make mistakes.
+The current transparent fee computation (inherited from Bitcoin) — input value
+less output value — is prone to user error. It is very easy to forget to add an
+output for a change address, make a calculation error [#bitcointalk-fee-error]_,
+mix up units etc.
+
+In addition, when signing a transaction with a hardware wallet, using an implicit 
+fee means that the hardware wallet must recompute the fee on its own and cannot
+simply display the value being committed to.
+
```

Comment:
Another important thing I just remembered: this will obviate the need for light clients to download all of the inputs to a transaction to determine the fee.